seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/sink/file/ClickhouseFileSinkFactory.java

```diff
@@ -60,4 +81,91 @@ public OptionRule optionRule() {
+            shardKeyType = tableSchema.get(shardKey);
+        }
+
+        ShardMetadata shardMetadata =
+                new ShardMetadata(
+                        shardKey,
+                        shardKeyType,
+                        readonlyConfig.get(DATABASE),
+                        readonlyConfig.get(TABLE),
+                        table.getEngine(),
+                        true,
+                        new Shard(1, 1, nodes.get(0)),
+                        readonlyConfig.get(USERNAME),
+                        readonlyConfig.get(PASSWORD));
+        List<String> fields = new ArrayList<>(tableSchema.keySet());
+
+        Map<String, String> nodeUser =
+                readonlyConfig.toConfig().getObjectList(NODE_PASS.key()).stream()
+                        .collect(
+                                Collectors.toMap(
+                                        configObject ->
+                                                configObject.toConfig().getString(NODE_ADDRESS),
+                                        configObject ->
+                                                configObject.toConfig().hasPath(USERNAME.key())
+                                                        ? configObject
+                                                                .toConfig()
+                                                                .getString(USERNAME.key())
+                                                        : "root"));
+
+        Map<String, String> nodePassword =
+                readonlyConfig.get(NODE_PASS).stream()
+                        .collect(
+                                Collectors.toMap(
+                                        NodePassConfig::getNodeAddress,
+                                        NodePassConfig::getPassword));
+
+        proxy.close();
+
+        if (readonlyConfig.get(FILE_FIELDS_DELIMITER).length() != 1) {
+            throw new ClickhouseConnectorException(
+                    SeaTunnelAPIErrorCode.CONFIG_VALIDATION_FAILED,
+                    FILE_FIELDS_DELIMITER.key() + " must be a single character");
+        }
+        FileReaderOption readerOption =
+                new FileReaderOption(
+                        shardMetadata,
+                        tableSchema,
+                        fields,
+                        readonlyConfig.get(CLICKHOUSE_LOCAL_PATH),
+                        ClickhouseFileCopyMethod.from(readonlyConfig.get(COPY_METHOD).getName()),
+                        nodeUser,
+                        readonlyConfig.get(NODE_FREE_PASSWORD),
+                        nodePassword,
+                        readonlyConfig.get(COMPATIBLE_MODE),
+                        readonlyConfig.get(FILE_TEMP_PATH),
+                        readonlyConfig.get(FILE_FIELDS_DELIMITER),
+                        readonlyConfig.get(KEY_PATH));
+
+        readerOption.setSeaTunnelRowType(catalogTable.getSeaTunnelRowType());
+        return () -> new ClickhouseFileSink(readerOption);
```

Comment:
When implement `createSink` method, we can delete the `prepare` method in `ClickhouseSink`